src/elements/content-explorer/__tests__/RenameDialog.test.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import userEvent from '@testing-library/user-event';
+
+import { render, screen } from '../../../test-utils/testing-library';
+import RenameDialog from '../RenameDialog';
+import { ERROR_CODE_ITEM_NAME_TOO_LONG, ERROR_CODE_ITEM_NAME_IN_USE } from '../../../constants';
+
+jest.mock('react-modal', () => {
+    return jest.fn(({ children }) => <div aria-label="Rename">{children}</div>);
+});
+
+const defaultProps = {
+    appElement: document.createElement('div'),
+    errorCode: '',
+    isLoading: false,
+    isOpen: true,
+    item: { name: 'test.txt', extension: 'txt' },
+    onCancel: jest.fn(),
+    onRename: jest.fn(),
+    parentElement: document.createElement('div'),
+};
+
+describe('elements/content-explorer/RenameDialog', () => {
+    const renderComponent = (props = {}) => render(<RenameDialog {...defaultProps} {...props} />);
+
+    test('renders the dialog with the correct initial state', () => {
+        renderComponent();
+        expect(screen.getByText('Please enter a new name for test:')).toBeInTheDocument();
+        expect(screen.getByRole('textbox')).toHaveValue('test');
+        expect(screen.getByDisplayValue('test')).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Rename' })).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+    });
+
+    test('calls onCancel when cancel button is clicked', async () => {
+        renderComponent();
+        await userEvent.click(await screen.findByRole('button', { name: 'Cancel' }));
```

Comment:
a lot of these seem to use `find` but why not use `get`? the buttons should appear on initial render and then the `await` isn't needed. that way there's no confusion about whether there's a delay or something in the rendering of this component